iroha/src/block_sync.rs

```diff
@@ -64,31 +64,36 @@ impl BlockSynchronizer {
         let kura = self.kura.clone();
         let peer_id = self.peer_id.clone();
         let sumeragi = self.sumeragi.clone();
-        drop(task::spawn(async move {
-            loop {
-                task::sleep(gossip_period).await;
-                let message =
-                    Message::LatestBlock(kura.read().await.latest_block_hash(), peer_id.clone());
-                drop(
-                    futures::future::join_all(
-                        sumeragi
-                            .read()
-                            .await
-                            .network_topology
-                            .sorted_peers()
-                            .iter()
-                            .map(|peer| message.clone().send_to(peer)),
-                    )
-                    .await,
-                );
+        drop(task::spawn(
+            async move {
+                loop {
+                    task::sleep(gossip_period).await;
+                    let message = Message::LatestBlock(
+                        kura.read().await.latest_block_hash(),
+                        peer_id.clone(),
+                    );
+                    drop(
+                        futures::future::join_all(
+                            sumeragi
+                                .read()
+                                .await
+                                .network_topology
+                                .sorted_peers()
+                                .iter()
+                                .map(|peer| message.clone().send_to(peer)),
+                        )
+                        .await,
+                    );
+                }
             }
-        }));
+            .in_current_span(),
```

Comment:
Calling this method allow to log something in another thread with the same span as in current thread, am I right?

Reply:
Yes!